### ISSUE TITLE
Unable to call context.read with provider of nullable variable

### ISSUE THREAD

**Issue by AlexHartford** [rrousselGit/river_pod#372]
context.read, context.refresh, etc. are unable to be called with param of a provider of a nullable variable. It seems to work for primitives but nothing else.

Error: The argument type 'Provider<User?>' can't be assigned to the parameter type 'ProviderBase<Object, dynamic>'.

```dart
final fAuthStream = StreamProvider<User?>((ref) => ref.watch(fAuth).authStateChanges());

final user = Provider<User?>((ref) => ref.watch(fAuthStream).data?.value);

Route<dynamic> generateRoute(RouteSettings settings) {
  switch (settings.name) {
    ...
    default:
      return MaterialPageRoute<void>(
        // error
        builder: (context) => context.read(user) == null ? Landing() : Home(),
      );
  }
}
```

However, if I modify the code to provide a non-nullable param, there is no error.

```dart
final user = Provider<User>((ref) => User());

Route<dynamic> generateRoute(RouteSettings settings) {
  switch (settings.name) {
    ...
    default:
      return MaterialPageRoute<void>(
        // works
        builder: (context) => context.read(user) == null ? Landing() : Home(),
      );
  }
}
```

But obviously with null safety and streams, we are working with nullables a lot. Maybe we need a context.maybeRead that accepts `T?`?

**Comment by syklevin** [comment]
I can't update to new version due to the same issue

**Comment by MarcinusX** [comment]
Got the same thing. It works with `StateProvider` though so maybe replacing `Provider` with `StateProvider` can be a temporary workaround.

**Comment by rrousselGit** [comment]
I'll make a fix later this week

**Comment by AlexHartford** [comment]
Went ahead and made PR that fixes the issue for me.
https://github.com/rrousselGit/river_pod/pull/373

**Comment by rrousselGit** [comment]
Fixed by #377 / #373
